### PR TITLE
424: Description updates on when to exclude to avoid double counting List/Contracted

### PR DESCRIPTION
**Pull request by udam-f2**
Issue #424

### CODE REVIEW

**Review by mike-finopsorg — APPROVED** [review]

File: specification/columns/contractedcost.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 Contracted Cost represents the cost calculated by multiplying [*contracted unit price*](#glossary:contracted-unit-price) and the corresponding [Pricing Quantity](#pricingquantity). Contracted Cost is denominated in the [Billing Currency](#billingcurrency) and is commonly used for calculating savings based on negotiation activities, by comparing it with [List Cost](#listcost). If negotiated discounts are not applicable, the Contracted Cost defaults to the List Cost.
 
+When aggregating for savings calculations, it's important to exclude one-time or recurring charges that are paid to cover future eligible charges or the covered charges themselves. This exclusion helps prevent double counting of these charges in the aggregation. For instance, charges categorized as [ChargeCategory](#chargecategory) "Purchase" and their related [ChargeCategory](#chargecategory) "Tax" charges for a [Commitment-Based Discount](#glossary:commitment-based-discount) might be excluded from an accrual-basis cost aggregation of Contracted Cost. This is because the "Usage" and "Tax" charge records provided during the term of the commitment discount already specify the Contracted Cost.
```
Comment (AWS-ZachErdman) [inline specification/columns/contractedcost.md:5]:
```suggestion
When aggregating for savings calculations, it's important to exclude either one-time or recurring charges ([ChargeCategory](#chargecategory)"Purchase") that are paid to cover future eligible charges or the covered charges themselves. Which you exclude depends on if you are aggregating cost on a billed basis (exclude Purchases) or accrual basis (exclude covered charges). This exclusion helps prevent double counting of these charges in the aggregation. For instance, charges categorized as [ChargeCategory](#chargecategory) "Purchase" and their related [ChargeCategory](#chargecategory) "Tax" charges for a [Commitment-Based Discount](#glossary:commitment-based-discount) might be excluded from an accrual-basis cost aggregation of Contracted Cost. This is because the "Usage" and "Tax" charge records provided during the term of the commitment discount already specify the Contracted Cost.
```

File: specification/columns/listcost.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
+List Cost represents the cost calculated by multiplying the [*list unit price*](#glossary:list-unit-price) and the corresponding [Pricing Quantity](#pricingquantity). List Cost is denominated in the [Billing Currency](#billingcurrency) and is commonly used for calculating savings based on various rate optimization activities, by comparing it with [Contracted Cost](#contractedcost), [Billed Cost](#billedcost) and [Effective Cost](#effectivecost).
+
+When aggregating for savings calculations, it's important to exclude one-time or recurring charges that are paid to cover future eligible charges or the covered charges themselves. This exclusion helps prevent double counting of these charges in the aggregation. For instance, charges categorized as [ChargeCategory](#chargecategory) "Purchase" and their related [ChargeCategory](#chargecategory) "Tax" charges for a [Commitment-Based Discount](#glossary:commitment-based-discount) might be excluded from an accrual-basis cost aggregation of List Cost. This is because the "Usage" and "Tax" charge records provided during the term of the commitment discount already specify the List Cost.
```
Comment (AWS-ZachErdman) [inline specification/columns/listcost.md:5]:
```suggestion
When aggregating for savings calculations, it's important to exclude either one-time or recurring charges ([ChargeCategory](#chargecategory)"Purchase") that are paid to cover future eligible charges or the covered charges themselves. This exclusion helps prevent double counting of these charges in the aggregation. Which you exclude depends on if you are aggregating cost on a billed basis (exclude Purchases) or accrual basis (exclude covered charges). For instance, charges categorized as [ChargeCategory](#chargecategory) "Purchase" and their related [ChargeCategory](#chargecategory) "Tax" charges for a [Commitment-Based Discount](#glossary:commitment-based-discount) might be excluded from an accrual-basis cost aggregation of List Cost. This is because the "Usage" and "Tax" charge records provided during the term of the commitment discount already specify the List Cost.
```

**Comment by AWS-ZachErdman** [comment]
Just added two suggestions to make it a little more clear. I was confused the first time I read the additions about whether it was saying I should exclude both the purchases and covered charges.

File: specification/columns/contractedcost.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
...
 
+When aggregating Contracted Cost for savings calculations, it's important to exclude either one-time or recurring charges ([ChargeCategory](#chargecategory)"Purchase") that are paid to cover future eligible charges or the covered charges themselves. This exclusion helps prevent double counting of these charges in the aggregation. Which set of charges to exclude depends on whether cost are aggregated on a billed basis (exclude covered charges) or accrual basis (exclude Purchases for future charges). For instance, charges categorized as [ChargeCategory](#chargecategory) "Purchase" and their related [ChargeCategory](#chargecategory) "Tax" charges for a [Commitment-Based Discount](#glossary:commitment-based-discount) might be excluded from an accrual basis cost aggregation of Contracted Cost. This is because the "Usage" and "Tax" charge records provided during the term of the commitment discount already specify the Contracted Cost.
```
Comment (flanakin) [inline specification/columns/contractedcost.md:5]: If we're going to require people to filter rows out in order to use some of our most important columns, we need a prescriptive way for them to apply that filter. If we don't have that, then I would lean towards not enabling such a non-obvious, breaking scenario.

File: supporting_content/columns/contractedcost.md [hunk 4]
```diff
@@ -36,6 +36,12 @@ Current column mappings found in available data sets:
 
 ## Discussion / Scratch space
 
+* 4/20/2024 update: Double counting issue was discussed in github issue [#424](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/424). Options discussed were:
```
Comment (flanakin) [inline supporting_content/columns/contractedcost.md:39]: nit: We should spell out the date to be more globally inclusive.
```suggestion
* April 20, 2024 update: Double counting issue was discussed in github issue [#424](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/424). Options discussed were:
```

File: supporting_content/columns/listcost.md [hunk 5]
```diff
@@ -47,3 +47,11 @@ Current column mappings found in available data sets:
 ### Current data sources
 
 * Even though some major CSPs provide candidate columns for mapping to ListCost within their billing data, it is advisable to rely on multiplying the predetermined values in the FOCUS columns ListUnitPrice and PricingQuantity. This approach ensures more consistent and reliable results.
+
+### Avoid double counting
+
+* 4/20/2024 Double counting issue was discussed in github issue [#424](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/424). Options discussed were:
```
Comment (flanakin) [inline supporting_content/columns/listcost.md:53]:
```suggestion
* April 20, 2024 Double counting issue was discussed in github issue [#424](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/424). Options discussed were:
```

File: specification/columns/contractedcost.md (hunk 3 above)
Comment (AWS-ZachErdman) [reply specification/columns/contractedcost.md:5]: I believe this double counting will also happen with the proposed `ActualQuantity` column which shows purchase quantities and actual usage quantities in the same column. So you'd similarly need to filter when aggregating on `ActualQuantity` for either commitment purchases or commitment usage.

I think it's just a consequence of showing two directions of transactions with the same columns.

Comment (flanakin) [inline specification/columns/contractedcost.md:5]: Can you rephrase "billed basis" and "accrual basis" to be more inclusive? Maybe explain the scenario in natural language without jargon and include `(also known as "billed basis")`?

Comment (jpradocueva) [reply specification/columns/contractedcost.md:5]: This conversation was resolved by the group, pending a new issue/PR to be raised by Michael.

**Review by jpradocueva — APPROVED** [review]
This PR was approved by the group. Michael will raise a new item to address some of the concerns expressed during the discussion.

File: specification/columns/contractedcost.md (hunk 3 above)
Comment (UdamDewaraja) [reply specification/columns/contractedcost.md:5]: In the past we've said terms that are not industry terms and can be easily googled don't need special treatment. We used the same logic when deciding what's needed for the glossary and what can be skipped. Happy to look at a suggestion but my personal view is these are pretty standard terms that a quick Google brings up if anyone isn't sure.

**Review by rileyjenk — APPROVED** [review]

**Review by ijurica — APPROVED** [review]

File: specification/columns/contractedcost.md (hunk 3 above)
Comment (cnharris10) [inline specification/columns/contractedcost.md:5]:
```suggestion
When aggregating Contracted Cost for savings calculations, ==it's important to exclude either one-time or recurring charges ([ChargeCategory](#chargecategory)"Purchase")== that are paid to cover future eligible charges or the covered charges themselves. This exclusion helps prevent double counting of these charges in the aggregation. Which set of charges to exclude depends on whether cost are aggregated on a billed basis (exclude covered charges) or accrual basis (exclude Purchases for future charges). For instance, charges categorized as [ChargeCategory](#chargecategory) "Purchase" and their related [ChargeCategory](#chargecategory) "Tax" charges for a [Commitment-Based Discount](#glossary:commitment-based-discount) might be excluded from an accrual basis cost aggregation of Contracted Cost. This is because the "Usage" and "Tax" charge records provided during the term of the commitment discount already specify the Contracted Cost.
```

Comment (jpradocueva) [reply specification/columns/contractedcost.md:5]: The group agreed to format the statement to be highlighted.

Comment (jpradocueva) [reply specification/columns/contractedcost.md:5]: Udam will create a new ticket for 1.1 and this will be resolved.

Comment (udam-f2) [reply specification/columns/contractedcost.md]: Updated to emphasize the double counting warning paragraph.
